docs/frc/1-Basics/9-Drivetrain.md

```diff
@@ -1,3 +1,300 @@
+
+## Subsystem File
+
+### Declarations
+
+While we have 6 wheels on each side, the 3 wheels are all connected to each other and must run in the same direction (which makes sense or else we would have wheels running in opposite directions). While we could just use one motor on both side of our robot, we actually use two per side (one labelled **front** and the other labelled **back**) to add extra power to our drivetrain. To begin our subsystem file, we might declare these motors as `frontLeftMotor`, `frontRightMotor`, `backLeftMotor`, and `backRightMotor`:
+
+With this tank drive design, going backwards and forwards is very easy: we simply set all the motors to full speed forwards or backwards. However, turning gets a bit more complicated. Instead, we have to spin the two sides of the drivetrain in opposite directions. For instance, to turn right, we would have to spin the right wheels backwards and the left wheels forwards. This would do a rotation about the center of our robot.
+
+The above doesn't seem that bad to program until we get to the fact that driving forward and turning at the same time can be a bit challenging. While the code isn't that bad to write from scratch, our implementation can be buggy at first and be pretty tedious to polish. 
+
+Luckily, WPILib makes this much easier for us by introducing a class called `DifferentialDrive`. What this class does is introduce a way to control our motors for us and do the math to turn what we want into the individual motor outputs. Turning while driving (not turning in place) is essentially using mismatched speeds with respect to the two sides. 
+
+We can pass in values (i.e. inputs from the controller's joysticks) into a `DifferentialDrive` object in order to tell the robot to drive forward at full speed while turning right at half speed. We'll name those passed-in values `speedForward` and `speedTurn`.
+
+Then, `DifferentialDrive` will take in these commands, do the math, and then update the motors accordingly to make the robot perform this action. 
+
+> [!NOTE]
+> Click [here](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/drive/DifferentialDrive.java) for `DifferentialDrive`'s source code.
+
+This is what we have so far in our file, with all the appropriate import statements:
+
+```java
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+
+public class Drivetrain extends SnailSubsystem {
+
+    private final CANSparkMax frontLeftMotor;
+    private final CANSparkMax frontRightMotor;
+    private final CANSparkMax backLeftMotor;
+    private final CANSparkMax backRightMotor;
+
+    private final DifferentialDrive drivetrain;
+
+    private double speedForward;
```

Comment:
You don't explain these at all and what I like to do (and what we did for Arm and ELevator) is actually consider these as sort of "extensions" of the state. We use the `state` variable as sort of an "operating mode" while the `speed` variables represent more information about our state and current operating behavior.